src/openssl_auth.cpp

```diff
@@ -96,14 +95,6 @@ static void s3fs_crypt_mutex_lock(int mode, int pos, const char* file, int line)
     }
 }
 
-static unsigned long s3fs_crypt_get_threadid() __attribute__ ((unused));
-static unsigned long s3fs_crypt_get_threadid()
-{
-    // For FreeBSD etc, some system's pthread_t is structure pointer.
-    // Then we use cast like C style(not C++) instead of ifdef.
-    return (unsigned long)(pthread_self());
-}
-
```

Comment:
This function is necessary.
However, it requires changes to std::thread as follows:
```
static unsigned long s3fs_crypt_get_threadid() __attribute__ ((unused));
static unsigned long s3fs_crypt_get_threadid()
{
    std::hash<std::thread::id> hash;
    return (unsigned long)hash(std::this_thread::get_id());
}
```

Reply:
I'm not sure this is actually used in OpenSSL 1.0 and later: https://linux.die.net/man/3/crypto_set_id_callback but I added it as you suggested.

Reply:
Thank you for adding it.
I'm also concerned, but there's the datarace issue that's also mentioned in other issues, so I'd like to keep it.